cirq-core/cirq/ops/conditional_operation.py

```diff
@@ -0,0 +1,155 @@
+        return ConditionalOperation(self._sub_operation.with_qubits(*new_qubits), self._controls)
+
+    def _decompose_(self):
+        result = protocols.decompose_once(self._sub_operation, NotImplemented)
+        if result is NotImplemented:
+            return NotImplemented
+
+        return [ConditionalOperation(op, self._controls) for op in result]
+
+    def _value_equality_values_(self):
+        return (frozenset(self._controls), self._sub_operation)
+
+    def __str__(self) -> str:
+        return repr(self)
+
+    def __repr__(self):
+        return f'ConditionalOperation({self._sub_operation!r}, {list(self._controls)!r})'
+
+    def _is_parameterized_(self) -> bool:
+        return protocols.is_parameterized(self._sub_operation)
+
+    def _parameter_names_(self) -> AbstractSet[str]:
+        return protocols.parameter_names(self._sub_operation)
+
+    def _resolve_parameters_(
+        self, resolver: 'cirq.ParamResolver', recursive: bool
+    ) -> 'ConditionalOperation':
+        new_sub_op = protocols.resolve_parameters(self._sub_operation, resolver, recursive)
+        return ConditionalOperation(new_sub_op, self._controls)
+
+    def __pow__(self, exponent: Any) -> 'ConditionalOperation':
+        new_sub_op = protocols.pow(self._sub_operation, exponent, NotImplemented)
+        if new_sub_op is NotImplemented:
+            return NotImplemented  # coverage: ignore
+        return ConditionalOperation(new_sub_op, self._controls)
+
+    def _circuit_diagram_info_(
+        self, args: 'cirq.CircuitDiagramInfoArgs'
+    ) -> Optional['protocols.CircuitDiagramInfo']:
+        sub_args = protocols.CircuitDiagramInfoArgs(
+            known_qubit_count=args.known_qubit_count,
+            known_qubits=args.known_qubits,
+            use_unicode_characters=args.use_unicode_characters,
+            precision=args.precision,
+            qubit_map=args.qubit_map,
+        )
+        sub_info = protocols.circuit_diagram_info(self._sub_operation, sub_args, None)
+        if sub_info is None:
+            return NotImplemented  # coverage: ignore
+
+        wire_symbols = sub_info.wire_symbols + ('^',) * len(self._controls)
+        exponent_qubit_index = None
+        if sub_info.exponent_qubit_index is not None:
+            exponent_qubit_index = sub_info.exponent_qubit_index + len(self._controls)
+        elif sub_info.exponent is not None:
+            exponent_qubit_index = len(self._controls)
+        return protocols.CircuitDiagramInfo(
+            wire_symbols=wire_symbols,
+            exponent=sub_info.exponent,
+            exponent_qubit_index=exponent_qubit_index,
+        )
+
+    def _json_dict_(self) -> Dict[str, Any]:
+        return {
+            'cirq_type': self.__class__.__name__,
+            'controls': self._controls,
+            'sub_operation': self._sub_operation,
+        }
+
+    def _act_on_(self, args: 'cirq.ActOnArgs') -> bool:
+        def not_zero(measurement):
```

Comment:
Actually I'm thinking `_controls` should be `List[Tuple[MeasurementKey, int], ...]` instead, allowing users to specify the index to control on. This would also make compatibility with QASM easier.

Reply:
[No change required] This function is kind of unnecessary in this PR, given that it's a one-liner, but I can see it becoming worthwhile once we start adding support for non-1 conditions.